jupyter_scheduler/models.py

```diff
@@ -11,6 +11,36 @@
 SCHEDULE_RE = ""
 
 
+class NotificationEvent(str, Enum):
+    """
+    Enum that represents events triggering notifications. Implementers can extend
+    this enum to include additional notification events as needed.
+
+    Attributes:
+        SUCCESS (str): Sent when a job completes successfully.
+        FAILURE (str): Sent on job failure.
+        STOPPED (str): Sent when a job is manually stopped.
+    """
+
+    SUCCESS = "Success"
+    FAILURE = "Failure"
+    STOPPED = "Stopped"
+
+
+class Notification(BaseModel):
```

Comment:
I thought about this + potential confusion with [Notification class in JupyterLab](https://github.com/jupyterlab/jupyterlab/blob/main/jupyterlab/handlers/announcements.py) but still decided to opt for a short name that is clear enough given it's use within Scheduler extension (+ I thought alternatives I came up with were not adding much clarity).

I'm ok to use a more verbose name for the sake of clarity but let's consider options. Problem with `ScheduledNotifications` is that only notebook execution is scheduled. We don't schedule notifications, we define to whom and based on which event/status change they should be sent. `PendingNotifications` to me means a batch of notification objects that are waiting to be sent.

Taking this all into account, `NotificationsSettings` seems like a good name because it does a good job explaining what this class is (necessary data / settings / configuration to send multiple notifications later). What do you think?

Reply:
`NotificationConfig` would be good. 👍